lib/releaselib.sh

```diff
@@ -449,15 +435,9 @@ release::set_release_version () {
         RELEASE_VERSION[rc]="v${build_version[major]}.${build_version[minor]}"
         RELEASE_VERSION[rc]+=".$((${build_version[patch]}+1))-rc.0"
       fi
-    elif [[ "${build_version[label]}" == "-rc" ]] || [[ "$FLAGS_type" == rc ]]; then
-      # betas not allowed after release candidates
+    elif [[ "$FLAGS_type" == rc ]]; then
       RELEASE_VERSION[rc]="$RELEASE_VERSION_PRIME"
-      if [[ "${build_version[label]}" == "-rc" ]]; then
-        RELEASE_VERSION[rc]+="-rc.$((${build_version[labelid]}+1))"
-      else
-        # Start release candidates at 1 instead of 0
-        RELEASE_VERSION[rc]+="-rc.1"
-      fi
+      RELEASE_VERSION[rc]+="-rc.$((${build_version[labelid]}+1))"
       RELEASE_VERSION_PRIME="${RELEASE_VERSION[rc]}"
     fi
   elif [[ "$FLAGS_type" == "beta" ]]; then
```

Comment:
Theoretically, if this code path gets executed during the beta cut, then we should get the correct version, right?